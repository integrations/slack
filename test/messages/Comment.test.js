// contents.json combined with params from github-url?

const { Comment } = require('../../lib/messages/comment');
const repository = require('../fixtures/repo.json');
const comment = require('../fixtures/comment.json');
const issue = require('../fixtures/issue.json');

describe('Comment rendering', () => {
  test('works for full unfurls', async () => {
    const blobMessage = new Comment({
      comment,
      repository,
      issue,
      unfurlType: 'full',
    });
    const rendered = blobMessage.getRenderedMessage();
    expect(rendered).toMatchSnapshot();
  });

  test('works for condensed unfurls', async () => {
    const blobMessage = new Comment({
      comment,
      repository,
      issue,
      unfurlType: 'condensed',
    });
    const rendered = blobMessage.getRenderedMessage();
    expect(rendered).toMatchSnapshot();
  });

  test('converts HTML body to mrkdwn', () => {
    const message = new Comment({
      comment: Object.assign({
        body_html: '<strong>Hello</strong> <em>cruel</em> <a href="http://example.com">world</a>',
      }, comment),
      repository,
      issue,
      unfurlType: 'full',
    });

    expect(message.getRenderedMessage().text).toEqual('*Hello* _cruel_ <http://example.com|world>');
  });

  test('trims HTML body after hr', () => {
    const message = new Comment({
      comment: Object.assign({
        body_html: '<strong>Hello</strong> <hr> <em>cruel</em> <a href="http://example.com">world</a>',
      }, comment),
      repository,
      issue,
      unfurlType: 'full',
    });

    expect(message.getRenderedMessage().text).toEqual('*Hello* ');
  });

  test('extracts image from body_html', () => {
    const message = new Comment({
      comment: Object.assign({
        body_html: '<p>Hello world!</p><p><img src="https://media.giphy.com/media/5xtDarEbygs3Pu7p3jO/giphy.gif"></p>',
      }, comment),
      repository,
      issue,
      unfurlType: 'full',
    });

    expect(message.getRenderedMessage()).toMatchSnapshot();
  });
});
