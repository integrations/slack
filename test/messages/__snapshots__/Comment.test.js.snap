// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment rendering extracts image from body_html 1`] = `
Object {
  "author_icon": "https://avatars2.githubusercontent.com/u/887?v=4",
  "author_link": "https://github.com/mislav",
  "author_name": "mislav",
  "color": "#24292f",
  "fallback": "[bkeepers/dotenv] Comment on #10191",
  "footer": "<https://github.com/bkeepers/dotenv|bkeepers/dotenv>",
  "footer_icon": "https://github.githubassets.com/favicon.ico",
  "image_url": "https://media.giphy.com/media/5xtDarEbygs3Pu7p3jO/giphy.gif",
  "mrkdwn_in": Array [
    "text",
  ],
  "text": "Hello world!

https://media.giphy.com/media/5xtDarEbygs3Pu7p3jO/giphy.gif",
  "title": "Comment on #10191 Consider re-licensing to AL v2.0, as RocksDB has just done",
  "title_link": "https://github.com/github/hub/pull/1535#issuecomment-322500379",
  "ts": 1502810673,
}
`;

exports[`Comment rendering works for condensed unfurls 1`] = `
Object {
  "color": "#24292f",
  "fallback": "[bkeepers/dotenv] Comment on #10191",
  "footer": "<https://github.com/bkeepers/dotenv|bkeepers/dotenv>",
  "footer_icon": "https://github.githubassets.com/favicon.ico",
  "mrkdwn_in": Array [
    "text",
  ],
  "title": "Comment on #10191 Consider re-licensing to AL v2.0, as RocksDB has just done",
  "title_link": "https://github.com/github/hub/pull/1535#issuecomment-322500379",
  "ts": 1502810673,
}
`;

exports[`Comment rendering works for full unfurls 1`] = `
Object {
  "author_icon": "https://avatars2.githubusercontent.com/u/887?v=4",
  "author_link": "https://github.com/mislav",
  "author_name": "mislav",
  "color": "#24292f",
  "fallback": "[bkeepers/dotenv] Comment on #10191",
  "footer": "<https://github.com/bkeepers/dotenv|bkeepers/dotenv>",
  "footer_icon": "https://github.githubassets.com/favicon.ico",
  "mrkdwn_in": Array [
    "text",
  ],
  "text": "Thanks for your work on this! I will open a PR for \`bump-formula-pr\` and CC you on it for ðŸ‘€ .",
  "title": "Comment on #10191 Consider re-licensing to AL v2.0, as RocksDB has just done",
  "title_link": "https://github.com/github/hub/pull/1535#issuecomment-322500379",
  "ts": 1502810673,
}
`;
